enhancements/cluster-api/cluster-autoscaler-integration-with-openshift-cluster-api.md

```diff
@@ -0,0 +1,709 @@
+`scaleTargetRef` and the sync controller will be able to properly mirror the
+changes by detecting that the Cluster Autoscaler Operater has authored the changes.
+
+#### MachineAutoscaler Status Changes
+
+Another implementation detail of this enhancement that users will interact with is the
+addition of Kubernetes conditions to the MachineAutoscaler status field
+(i.e. `.status.conditions`).
+
+Historically, the MachineAutoscaler resource has not contained information to aid users in
+determining the state and health of that scaling group. With this enhancement however, it
+is possible that users might encounter issues when interacting with non-authoritative
+MachineSet resources. To improve the user experience, a conditions field will be added
+to the MachineAutoscaler status field (i.e. `.status.conditions`).
+
+There will be a single condition to indicate the ready status of the MachineAutoscaler. Under
+normal operation this condition will appear similar to this example:
+
+```yaml
+status:
+  conditions:
+    - lastTransitionTime: 2025-01-01T00:00:00Z
+      message: MachineAutoscaler ready for autoscaling
+      reason: MachineAutoscalerReady
+      status: "True"
+      type: Ready
+```
+
+In a case where the user has specified the same MachineSet in multiple MachineAutoscaler resources,
+the condition would appear as follows:
+
+```yaml
+status:
+  conditions:
+    - lastTransitionTime: 2025-01-01T00:00:00Z
+      message: targetScaleRef has multiple MachineAutoscaler owners
+      reason: MachineAutoscalerDuplicateTargetScaleRef
+      status: "False"
+      type: Ready
+```
+
+In this manner, errors in user configuration will be exposed through the MachineAutoscaler
+resource to assist users in diagnosing error states.
+
+#### MachineSet Sync Controller Changes
+
+The behavior of the MachineSet sync controller is another focus of implementation detail.
+Specifically, the MachineSet annotations and the replicas field of the spec
+(i.e. `.spec.replicas`). As mentioned previously, the MachineSet sync controller will
+use the managed fields metadata to know who, or what, has updated a field. Based on the field,
+and the author of the update, the sync controller will either propagate the change
+or synchronize with the authoritative resource.
+
+The updates that the MachineSet sync controller will watch for fall into a few different
+categories:
+
+* `.metadata.annotations` changes will be synced from the authoritative resource to the
+  non-authoritative except in these cases:
+  * The Cluster Autoscaler Operator has added the minimum and maximum size annotations, and ownership
+    annotation to a record. If the sync controller sees an update to these annotations on a
+    non-authoritative resource originating from the Cluster Autoscaler Operator, it will copy
+    that change to the authoritative resource if no MachineAutoscaler is referencing the
+    authoritative resource.
```

Comment:
Since we own the CAO, I think implementing this within CAO is likely the easier option.

The sync logic already syncs annotations across between machinesets, we just need to add special logic to handle well known annotations if they somehow end up converting from annotations to structured status

Reply:
ack, i can update this

Reply:
updated